src/test/functional/functional-tests.js

```diff
@@ -255,6 +255,53 @@ describe('functional tests', function() {
       client.fPutObject(bucketName, _100kbObjectName, tmpFileUpload,done)
     })
 
+    step(`statObject(bucketName, objectName, cb)_bucketName:${bucketName}, objectName:${_100kbObjectName}_`, done => {
+      client.statObject(bucketName, _100kbObjectName, (e, stat) => {
+        if (e) return done(e)
+        // As metadata is not provided and there is no file extension,
+        // we default to 'application/octet-stream' as per `probeContentType` function
+        if (stat.metaData && stat.metaData['content-type'] !== 'application/octet-stream') {
+          return done(new Error('content-type mismatch'))
+        }
+        done()
+      })
+    })
+
+    var tmpFileUploadWithExt = `${tmpDir}/${_100kbObjectName}.txt`
+    step(`fPutObject(bucketName, objectName, filePath, metaData, callback)_bucketName:${bucketName}, objectName:${_100kbObjectName}, filePath: ${tmpFileUploadWithExt}, metaData:${metaData}_`, done => {
+      fs.writeFileSync(tmpFileUploadWithExt, _100kb)
+      client.fPutObject(bucketName, _100kbObjectName, tmpFileUploadWithExt, metaData, done)
+    })
+
+    step(`statObject(bucketName, objectName, cb)_bucketName:${bucketName}, objectName:${_100kbObjectName}_`, done => {
+      client.statObject(bucketName, _100kbObjectName, (e, stat) => {
+        if (e) return done(e)
+        // As metadata is provided, even though we have an extension,
+        // the `content-type` should be equal what was declared on the metadata
+        if (stat.metaData && stat.metaData['content-type'] !== 'text/html') {
```

Comment:
Why is it `'text/html'` here and `'text/plain` in the next test?

Reply:
It is `'text/html'` here as that was the `content-type` specified in the `metaData` on line 98, and it is `'text/plain'` on the next test as we didn't provide any metaData when we uploaded the object and we need to automatically detect the content type from the `.txt` extension.

Reply:
Thanks, did not see that.